controller/users.go

```diff
@@ -98,7 +114,18 @@ func copyExistingKeycloakUserProfileInfo(existingProfile *login.KeycloakUserProf
 	if existingProfile.Username != nil {
 		keycloakUserProfile.Username = existingProfile.Username
 	}
-	return keycloakUserProfile
+	return keycloakUserProfile, nil
+}
+
+func (c *UsersController) getKeycloakProfileInformation(tokenString string, accountAPIEndpoint string) (*login.KeycloakUserProfileResponse, error) {
+
+	response, err := c.userProfileService.Get(tokenString, accountAPIEndpoint)
+	if err != nil {
+		log.Error(context.Background(), map[string]interface{}{
```

Comment:
context.Background is empty. I suggest you to propagate the ctx from the entrypoint of the controller across functions. Like that we could be able to track and extract values such as identity_id, request_id in case of an  error.